dao-impl/ebean-dao/src/main/java/com/linkedin/metadata/dao/EbeanLocalDAO.java

```
@@ -1214,6 +1214,7 @@ private static String constructCountAggregateSQLQuery(@Nonnull IndexCriterionArr
       }
       whereClause.append(" t").append(i).append(".aspect = ?");
       if (criterion.getPathParams() != null) {
+        validateConditionAndValue(criterion);
```

Probably missed it earlier, let's move this outside, we can check for `getParams` to be non-null in `validateConditionAndValue` method itself? This way you will fail early.

---

I think the `pathParams` can be null in the index criterion: https://github.com/linkedin/datahub-gma/blob/master/dao-api/src/main/pegasus/com/linkedin/metadata/query/IndexCriterion.pdl#L16

---

Yes but we can check for non-null value inside `validateConditionAndValue` method itself?

---

We can address this separately.